frameworklauncher/src/main/java/com/microsoft/frameworklauncher/webserver/StatusManager.java

```diff
@@ -123,8 +136,34 @@ private void pullStatus() throws Exception {
             frameworkName));
       }
     }
+  }
+
+  private Map<String, AggregatedFrameworkStatus> getReusableAggregatedFrameworkStatuses() throws Exception {
```

Comment:
>getReusableAggregatedFrameworkStatuses [](start = 49, length = 38)

by reusable do you mean immutable? why not just name it getImmutableAggregatedFrameworkStatuses

Reply:
1. Not all immutable AggregatedFrameworkStatus can be reused, such as when the corresponding requested FrameworkName and FrameworkVersion is deleted or changed.

2. Besides immutable, there maybe others can be reused in future.

3. Immutable is too low level inside this class, the class or the caller pullStatus only cares about what can be reused and skipped to pull from ZK.

---
In reply to: [184356194](https://github.com/Microsoft/pai/pull/534#discussion_r184356194) [](ancestors = 184356194)